alf/pretrained_models/model_adapters.py

```diff
@@ -189,6 +189,26 @@ def forward(self, input):
         raise NotImplementedError()
 
 
+@alf.configurable
+class EmbeddingAdapter(LoRA):
+    """Adapter for embedding layers.
+    """
+
+    @classmethod
+    def can_adapt(cls, m: nn.Module) -> bool:
+        return isinstance(m, nn.Embedding)
+
+    @classmethod
+    def _decompose_weight_dims(cls, m: nn.Module):
+        return m.num_embeddings, m.embedding_dim
+
+    def forward(self, input):
+        m = self._m[0]
+        return F.embedding(input, self._adapter_weight(), m.padding_idx,
```

Comment:
This is quite costly. `F.embedding(input, wB, ...) @ wA` is more efficient when the low rank mode is used.

Reply:
> This is quite costly. `F.embedding(input, wB, ...) @ wA` is more efficient when the low rank mode is used.

updated